### DISCUSSION TITLE
How to get pre-push hooks to work with Sync Changes in vscode

### DISCUSSION THREAD

**Discussion by Fy50167** [evilmartians/lefthook#862] [Q&A]
Hello,

I currently have a pre-push hook with runs a suite of e2e tests from Cypress. When I manually type git push into my terminal lefthook will run and run my cypress test suite. However, when I try using the Sync changes button in the source control panel of vs code, it just loads infinitely. At first I thought the load time was because it was also running cypress, but the loading goes on infinitely. Is there some way to make lefthook compatible with the sync changes button?

Additionally, when I run git push, my entire cypress test suite runs and passes successsfully. However, at it passes, instead of the push going through I'll get this message:⠧ waiting: cypressConnection to github.com closed by remote host.

Then it'll go back to saying waiting for cypress, but this time it will just load indefinitely. Does anybody know why my testing doesn't terminate once everything passes?

**Comment by mrexox** [comment]
Hey! I guess you're pushing to a remote via ssh protocol. You are probably hitting the timeout and SSH connection breaks because tests take some time to run.

You can configure your SSH client to fix this: https://stackoverflow.com/questions/60833006/connection-to-github-com-closed-by-remote-host-when-pushing

To answer other questions I need more context:
- What are the logs of VSCode running git hooks?
- Could you share your lefthook config?
